src/marketplace/MakeATankSaleView.js

```diff
@@ -0,0 +1,118 @@
+//@flow strict
+import * as React from 'react';
+import OptionClass from '../armory/OptionClass.js';
+import {getAllUsersTanks} from '../globalComponents/tankAPIIntegration.js';
+import {getUser} from '../globalComponents/userAPIIntegration.js';
+import {makeASale} from './marketPlaceAPIConnections.js';
+
+type Props = {||}; 
+type State = {|
+	userId: string,
+	salePrice: number,
+	tankBeingSoldId: string,
+	itemAmount: number,
+	tanksToSell: Array<OptionClass>,
+|};
+class MakeAComponentSaleView extends React.Component<Props, State> {
+
+	constructor() {
+		super();
+		this.state={
+			userId: '',
+			salePrice: 0,
+			tankBeingSoldId: '',
+			itemAmount: 0,
+			tanksToSell: [],
+		}
+		this.getUserID();
+	}
+
+	//this gets the user id
+	getUserID() : void {
+		const responsePromise = getUser();
+		responsePromise.then(
+			response => response.json().then(data => {
+				if (response.status !== 200) {
+					console.log(response.status);
+					console.log(data.msg);
+					console.log(data);
+					return data;
+				}
+				else {
+					const jsonObjectOfUser = data;
+					//set the users id
+					this.setState({userId:jsonObjectOfUser._id});
+					//gets the tanks that the user has currently to sell
+					this.getAllUsersTanksForSell();
+				}
+			})
+		).catch(
+			error => {
+				console.log('Couldnt connect to server!');
+				console.log(error);
+			}
+		);
+	};
+
+	//This gets all of the users tanks and then adds them to the dropdown
+	getAllUsersTanksForSell() : void {
+		const responsePromise = getAllUsersTanks();
+		responsePromise.then(
+			response => response.json().then(data => {
+				if (response.status !== 200) {
+					console.log(response.status);
+					console.log(data.msg);
+					console.log(data);
+					return data;
+				}
+				else {
+					const jsonObjectOfTanks = data;
+					const tankOptions = [new OptionClass('', '')];
+					//for every tank we will make a select option
+					for (const tank in jsonObjectOfTanks) {
+						tankOptions.push(new OptionClass(jsonObjectOfTanks[tank]._id, jsonObjectOfTanks[tank].tankName));
+					}
+					this.setState({tanksToSell : tankOptions});
+				}
+			})
+		).catch(
+			error => {
+				console.log('Couldnt connect to server!');
+				console.log(error);
+				return error;
+			}
+		);
+	};
+
+	//This will make a sale for a tank
+	makeASaleOfATank = ():void => {
+		const responsePromise = makeASale(this.state.userId, this.state.salePrice, this.state.tankBeingSoldId, 'tank', 1);
+		responsePromise.then(
+			response => response.json().then(data => {
+				console.log(data);
+				//lets get the new tanks that we have since we lost the current one
+				this.getAllUsersTanksForSell();
+				//set the new selling price to zero 
+				this.setState({salePrice: 0});
+				}
+			)
```

Comment:
}) should be on the same line so the indentation matches up

Reply:
Also, you are dealing with the backend here which could fail, so you need to handle what happens if it does.